chain/chain/src/flat_storage_resharder.rs

```diff
@@ -569,6 +576,10 @@ impl FlatStorageResharder {
                     parent_shard,
                     FlatStorageStatus::Ready(FlatStorageReadyStatus { flat_head }),
                 );
+                self.runtime
+                    .get_flat_storage_manager()
+                    .get_flat_storage_for_shard(parent_shard)
+                    .map(|flat_storage| flat_storage.set_flat_head_update_mode(true));
```

Comment:
Is it not needed in `FlatStorageReshardingTaskResult::Cancelled` case too?

Reply:
It's a good point you raised.
The operation of 'cancelling' resharing in the context of flat storage means that the node got SIGINT'ed. We want to cancel the tasks to shutdown quickly, but continue resharding on startup. So we keep flat storage locked to ensure the its state is locked in `Resharding`.

We could argue that unlocking flat storage is not needed also for `Failed` since we panic immediately, but well, maybe one day we won't panic